pkg/iosource/file.go

```diff
@@ -0,0 +1,30 @@
+package iosource
+
+import (
+	"io"
+	"net/url"
+	"os"
+)
+
+var DefaultFileSource = &FileSource{Perm: 0666}
+
+type FileSource struct {
+	Perm os.FileMode
+}
+
+func (f *FileSource) NewReader(path string) (io.ReadCloser, error) {
+	return os.Open(filePath(path))
+}
+
+func (f *FileSource) NewWriter(path string) (io.WriteCloser, error) {
+	path = filePath(path)
+	return os.OpenFile(path, os.O_RDWR|os.O_CREATE|os.O_TRUNC, f.Perm)
```

Comment:
I think you need to use pkg/fs.OpenFile here.

Reply:
Yep, good call.